config/routes.rb

```diff
@@ -1,6 +1,6 @@
 Rails.application.routes.draw do
-  # Define your application routes per the DSL in https://guides.rubyonrails.org/routing.html
-
-  # Defines the root path route ("/")
-  # root "articles#index"
+  root to: "users#index"
+  get "users/:id", to: "users#show", as: "user"
+  get "users/:author_id/:posts", to: "posts#index", as: "user_posts"
+  get "users/:author_id/:posts/:id", to: "posts#show", as: "user_post"
 end
```

Comment:
- [ ]  You have done a great job to implement  routes, but you still need to improve them to full fill the requirement.  You have not implemented the` post#show` routes, and I would recommend you to change the styles you've written your routes and use `resources` method .  using resources will easily help you to nest the routes and to have a very clean codes

Reply:
@ishimwezachee  thanks for tip, will put effort to implement it